tests/test_multi_inputport_modules_stage.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+# SPDX-FileCopyrightText: Copyright (c) 2022-2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+from unittest import mock
+
+import pytest
+
+from morpheus.stages.general.multi_inputport_modules_stage import MultiInputportModulesStage
+
+module_conf = {
+    "module_id": "TestModule", "module_name": "test_multi_inputport_module", "namespace": "test_morpheus_modules"
+}
+
+
+@pytest.mark.use_python
+def test_constructor(config):
+
+    mod_stage = MultiInputportModulesStage(config, module_conf=module_conf, num_input_ports_to_merge=2)
+
+    assert mod_stage.name == "test_multi_inputport_module"
+
+    # Just ensure that we get a valid non-empty tuple
+    accepted_types = mod_stage.accepted_types()
+    assert isinstance(accepted_types, tuple)
+    assert len(accepted_types) > 0
+    assert accepted_types[0] == typing.Any
+
+    assert mod_stage.supports_cpp_node() is False
+
+
+@pytest.mark.use_python
+def test_build_after_module_registration(config, register_test_module):
+
+    mock_node = mock.MagicMock()
+    mock_segment = mock.MagicMock()
+    mock_module = mock.MagicMock()
+    mock_input_stream = mock.MagicMock()
+
+    mock_segment.load_module.return_value = mock_module
+    mock_segment.make_node.return_value = mock_node
+
+    mod_stage = MultiInputportModulesStage(config, module_conf=module_conf, num_input_ports_to_merge=2)
+
+    mod_stage._build(mock_segment, mock_input_stream)
+
+    mock_segment.load_module.assert_called_once()
+    assert mock_segment.make_edge.call_count == 2
+
+
+@pytest.mark.use_python
+def test_invalid_number_input_ports(config, register_test_module):
+    num_input_ports_to_merge = 0
+    with pytest.raises(ValueError):
+        MultiInputportModulesStage(config, module_conf=module_conf, num_input_ports_to_merge=num_input_ports_to_merge)
```

Comment:
Also want a test for incorrect port names

Reply:
Still missing tests for incorrect names

Reply:
@mdemoret-nv I tried to test with `pipeline.run()` with incorrect port names, but it resulted in an `invalid ports` error which i can assert but it is followed by a segfault. Tests using mock_segment and mock_node to call `_build` function were successful because they are mock objects. Is there an other way to call the `_build` function?

Reply:
```
====================================================================================== 1 passed, 1 warning in 4.82s =======================================================================================
F20230516 18:19:25.776926 3928398 edge_holder.hpp:61] A node was destructed which still had dependent connections. Nodes must be kept alive while dependent connections are still active
*** Check failure stack trace: ***
    @     0x7fd483c7ef8d  google::LogMessage::Fail()
    @     0x7fd483c82e67  google::LogMessage::SendToLog()
    @     0x7fd483c7ea55  google::LogMessage::Flush()
    @     0x7fd483c7feaa  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fd4847bf57e  mrc::edge::EdgeHolder<>::~EdgeHolder()
    @     0x7fd4847c1fb9  mrc::pymrc::PythonNode<>::~PythonNode()
    @     0x7fd4847a30e7  mrc::segment::Runnable<>::~Runnable()
    @     0x7fd48479a622  std::_Rb_tree<>::_M_erase()
    @     0x7fd48479af6b  mrc::pymrc::PythonSegmentModule::~PythonSegmentModule()
    @     0x7fd480746eda  (unknown)
    @     0x7fd480791f27  (unknown)
    @     0x7fd4848c0700  (unknown)
    @     0x5623d7f4a8d8  frame_dealloc
    @     0x5623d7f5abca  tb_dealloc
    @     0x5623d7f5ac2e  tb_dealloc
    @     0x5623d7f5ac2e  tb_dealloc
    @     0x5623d7f5ac2e  tb_dealloc
    @     0x5623d7f5ac2e  tb_dealloc
    @     0x5623d7f5ac2e  tb_dealloc
    @     0x5623d7f5ac2e  tb_dealloc
    @     0x5623d7f5ac2e  tb_dealloc
    @     0x5623d7f5ac2e  tb_dealloc
    @     0x5623d7f5ac2e  tb_dealloc
    @     0x5623d7f604ea  BaseException_dealloc
    @     0x7fd4869f7eaa  FutureObj_clear
    @     0x7fd4869f7f30  TaskObj_clear
    @     0x5623d7f34795  gc_collect_main
    @     0x5623d7fe898e  gc_collect_with_callback
    @     0x5623d801f2a3  PyGC_Collect
    @     0x5623d801ea52  Py_FinalizeEx.localalias
    @     0x5623d801f578  Py_Exit
    @     0x5623d801c98b  handle_system_exit
Aborted (core dumped)

```

Reply:
You want to mock the module and call `_validate_ports` directly instead of mocking the builder and node objects